### PR TITLE
Update CONTRIBUTING.md for new theme

### LINKED ISSUE #1868: Update CONTRIBUTING.md for new theme and for maintainers

### PR DESCRIPTION
**Pull request by mattstratton**
Fixes #1868

This change, when implemented, will update the contributing guidelines based upon the new theme. 
This PR should not be merged until we release the new `devopsdays-theme` into this repo.

This change is open for comments/feedback from the @devopsdays/website team please :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -48,27 +50,31 @@ Now open `baseURL` in a browser and navigate to the content that you're editing
...
 ### Guidelines
 
-1. Code changes that affect the overall site will be reviewed only if they are in a separate pull request from any event-specific content. In short, don't add "giant template change" in the same PR as "here are some more sponsors" - if it affects *anything other than your event*, it should be in its own PR.
-1. We use [github issues](https://github.com/devopsdays/devopsdays-web/issues) to track work, so feel free to create new issues if you like (or read/comment on existing ones).
-1. If you are proposing a change that affects the overall site, and is not tied to an existing issue, please open a [new issue](https://github.com/devopsdays/devopsdays-web/issues) so that it can be discussed by the team, prior to submitting a pull request.
-
-
-## Advanced
-If you are going to be making changes to the overall functionality of the site, please keep the following in mind:
+1. Code changes that affect the overall site should be made only in the [devopsdays-theme](https://github.com/devopsdays/devopsdays-theme) repo. Theme changes will be made there, and when released, will be migrated to this repo.
```
Comment (bridgetkromhout) [inline CONTRIBUTING.md]: > migrated to

I would say "released to be used in" - migrate implies moving and _staying_.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 The technical details on how to set up [Hugo](https://gohugo.io/) (to see your local edits), and prepare a Pull Request for inclusion on the [Devopsdays](http://www.devopsdays.org/) website.
```
Comment (bridgetkromhout) [inline CONTRIBUTING.md]: s/The/This README contains the/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
...
 
+If you'd like to assist in contributing the the overall code of the website, please see [devopsdays/devopsdays-theme](https://github.com/devopsdays/devopsdays-theme).
```
Comment (bridgetkromhout) [inline CONTRIBUTING.md]: > contributing the the overall code

contributing to the code

("overall" is not very descriptive.)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -48,27 +50,31 @@ Now open `baseURL` in a browser and navigate to the content that you're editing
...
-
-
-## Advanced
-If you are going to be making changes to the overall functionality of the site, please keep the following in mind:
+1. Code changes that affect the overall site should be made only in the [devopsdays-theme](https://github.com/devopsdays/devopsdays-theme) repo. Theme changes will be made there, and when released, will be migrated to this repo.
+1. We use [github issues](https://github.com/devopsdays/devopsdays-theme/issues) to track work, so feel free to create new issues if you like (or read/comment on existing ones).
+1. If you are proposing a change that affects the overall site, and is not tied to an existing issue, please open a [new issue](https://github.com/devopsdays/devopsdays-theme/issues) so that it can be discussed by the team, prior to submitting a pull request.
 
-### Changes to content should be separate from overall functionality
+### Do not make changes to non-content files
```
Comment (bridgetkromhout) [inline CONTRIBUTING.md]: > Do not make changes to non-content files

This is more clear if you say "Only make changes to event content files"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -48,27 +50,31 @@ Now open `baseURL` in a browser and navigate to the content that you're editing
...
 "Content" means anything inside the `/content/...`, `/data/...`, or `/static/...` directories.
 
 Changes to content should be submitted as a separate PR from changes to site functionality. It would be additionally delightful if you label PR's for site functionality (such as `bug` or `enhancement`), but that's not required.
```
Comment (bridgetkromhout) [inline CONTRIBUTING.md]: >   Changes to content should be submitted as a separate PR from changes to site functionality. 

I would change this to "Changes to site-wide content should be submitted as a separate PR from changes to your individual event."

> It would be additionally delightful if you label PR's for site functionality (such as `bug` or `enhancement`), but that's not required.

I would cut this line because we aren't going to accept changes to functionality here.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -48,27 +50,32 @@ Now open `baseURL` in a browser and navigate to the content that you're editing
...
-- [SimpleLess](http://wearekiss.com/simpless) - all platforms, free.
-- [Less.js](http://lesscss.org/) - node application, free.
+### PR Tests
 
-The only place to make changes to the LESS files is in `/themes/devopsdays-responsive/static/site_variables.less` and `/themes/devopsdays-responsive/static/site.less`. Make sure your LESS compiler is compiling the output into `/themes/devopsdays-responsive/static/css/`. *DO NOT* make any changes to any other LESS file (do not directly modify the bootstrap LESS files, for example).
+The following tests run when a PR is submitted:
+1. [Travis](https://travis-ci.org/devopsdays/devopsdays-web/) - this is a basic test that confirms that the site can be built with Hugo on linux, and it runs an `html-min` gulp task which will identify if there is any invalid HTML in the site. This protects the final build, so if the Travis tests fail, please take a look as to why they failed.
```
Comment (phrawzty) [inline CONTRIBUTING.md]: These bullets don't appear to render out properly:
image

I'm not sure why though. :thinking:

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -48,27 +50,33 @@ Now open `baseURL` in a browser and navigate to the content that you're editing
...
+1. [Appveyor](https://ci.appveyor.com/project/DevOpsDays/devopsdays-web) - this again is a small test that builds Hugo on Windows, to ensure that no Windows-incompatible files have been included. If Appveyor tests fail, merge at your own discretion, based upon the failure reason.
+1. [Gitmagic](https://gitmagic.io/) - This is a bot that makes sure our pull requests are fashioned cleanly. See [contributing.json](https://github.com/devopsdays/devopsdays-web/blob/master/contributing.json) for a list of rules that we enforce.
+1. Netlify - This is a very useful test. It builds the site, and hosts an ephemeral preview version of it (viewable by clicking on the "details" link next to the test once it has turned green). It's pretty important to view this "deploy preview" if the PR has changed anything significant (adding a sponsor, etc, probably not...but changing content in a large way? Yes.)
```
Comment (phrawzty) [inline CONTRIBUTING.md]: Given that this is a list of four items - the first three of which are linked - it _looks_ like a mistake that Netlify isn't also linked. Maybe link it here and unlink it in the jfrazelle shout-out?

Comment (mattstratton) [inline CONTRIBUTING.md]: I changed it to a link to the specific view of the netlify builds, which is not public, so only folks who have access will see it.

**Comment by phrawzty** [comment]
`r+` from me.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -48,27 +50,33 @@ Now open `baseURL` in a browser and navigate to the content that you're editing
...
-### CSS changes are done with LESS
-Please do NOT make changes to any of the files in the `/themes/devopsdays-responsive/static/css` directory. These are files that are compiled via [LESS](http://lesscss.org/), and while your changes in there might work, they will be blown away by the LESS compiler at some point.
 
-If you want to make changes to CSS, you will need a LESS compiler on your system. [@mattstratton](https://github.com/mattstratton) uses [CodeKit](https://incident57.com/codekit/), but that is OS X only, and is not free. Here's a list of other possible LESS compilers (note - Matt has not tested any of these):
-- [SimpleLess](http://wearekiss.com/simpless) - all platforms, free.
-- [Less.js](http://lesscss.org/) - node application, free.
+## Maintainer Guidlines
+
+If you have permissions to merge PR's on this repo, here are a few guidelines to watch out for:
+
```
Comment (MikeRosTX) [inline CONTRIBUTING.md]: Just a thought, but I think it would be good to have in this section of the **Maintainer Guidelines** something like:

- If you're reviewing all the details of a PR before merging or are communicating with the _Submitter_, add yourself to _Assignees_ so that others know someone is waiting on a response or reviewing all the details of the PR thoroughly.

- 

-

Comment (MikeRosTX) [inline CONTRIBUTING.md]: Just a thought, but I think it would be good to have in this section of the **Maintainer Guidelines** something like:

- If you're reviewing all the details of a PR before merging or are communicating with the _Submitter_, add yourself to _Assignees_ so that others know someone is waiting on a response or reviewing all the details of the PR thoroughly.

- 

-

**Comment by mattstratton** [comment]
@mikerostx I agree, but we need to think it through since the mention-bot will be auto-assigning some PR's as well, so just because someone is assigned doesn't mean they are looking at it :)

I think that you would definitely add yourself as assigned , but we would want some comment that you were reviewing it, etc. Also probably something like "add the Do-not-merge label if you are in working something out with the  submitter"

**Comment by MikeRosTX** [comment]
@mattstratton, ah gotcha on the bot.  Is there anyway that if a bot assigns, we can decline it if we wanted to?  Just trying to understand the function of the bot auto-assigning.

Definitely agree adding one self as a **Assignee**, followed up with comments and selecting **do-not-merge** (not necessarily in the title) to **Labels**.

**Comment by mattstratton** [comment]
@MikeRosTX you can always remove yourself as an assignee, after its been done, to the best of my knowledge (that's just plain old GitHub at that point).

The bot will ONLY assign it to someone if it's a "sensitive" file, such as `config.toml`, `.gitignore`, or any of the root content (`content/page/*`), so for most event-related pr's there won't be anyone assigned anyway.
